pilot/pkg/proxy/envoy/v2/mesh.go

```diff
@@ -0,0 +1,990 @@
+	// form. See xdsapi.Metadata.Name
+	istioConfigFilter = "io.istio.istio-config"
+
+	// TODO: move all DestinationAttributes to github.com/istio.io/api
+	// Key Istio attribute names for mapping endpoints to subsets.
+	// For details, please see
+	// https://istio.io/docs/reference/config/mixer/attribute-vocabulary.html
+
+	// DestinationUID is the attribute name for the mesh unique, platform-specific
+	// unique identifier for the server instance of the destination service. Mesh
+	// uses the label to uqniquely identify the endpoint when it receives updates
+	// from the Controller. Example: kubernetes://my-svc-234443-5sffe.my-namespace
+	// No two instances running anywhere in the mesh can have the same value for
+	// UID.
+	DestinationUID DestinationAttribute = "destination.uid"
+
+	// DestinationService represents the fully qualified name of the service that the server
+	// belongs to. Example: "my-svc.my-namespace.svc.cluster.local"
+	DestinationService DestinationAttribute = "destination.service"
+
+	// DestinationName represents the short name of the service that the server
+	// belongs to. Example: "my-svc"
+	DestinationName DestinationAttribute = "destination.name"
+
+	// DestinationNamespace represents the namespace of the service. Example: "default"
+	DestinationNamespace DestinationAttribute = "destination.namespace"
+
+	// DestinationDomain represents the domain portion of the service name, excluding
+	// the name and namespace, example: svc.cluster.local
+	DestinationDomain DestinationAttribute = "destination.domain"
+
+	// DestinationIP represents the IP address of the server instance, example 10.0.0.104.
+	// This IP is expected to be reachable from __THIS__ pilot. No distinction
+	// is being made for directly reachable service instances versus those
+	// behind a VIP. Istio's health discovery service will ensure that this
+	// endpoint's capacity is correctly reported accordingly.
+	DestinationIP DestinationAttribute = "destination.ip"
+
+	// DestinationPort represents the recipient port on the server IP address, Example: 443
+	DestinationPort DestinationAttribute = "destination.port"
+
+	// DestinationUser represents the user running the destination application, example:
+	// my-workload-identity
+	DestinationUser DestinationAttribute = "destination.user"
+
+	// DestinationProtocol represents the protocol of the connection being proxied, example:
+	// grpc
+	DestinationProtocol DestinationAttribute = "context.protocol"
+
+	// nameValueSeparator is a separator for creating label name-value keys used for
+	// reverse lookups on labels.
+	nameValueSeparator = "\x1F"
+
+	// RuleSubsetSeparator separates the destination rule name and the subset name for
+	// the key to the subsetEndpoints map.
+	RuleSubsetSeparator = "|"
+
+	// wildCardDomainPrefix is used to check if rule names are wild card domains. If
+	// the rule name is prefixed with this pattern, consider it a wild card.
+	wildCardDomainPrefix = "*."
```

Comment:
This does not align with how we perform wild card checking elsewhere (nor does it align with how Envoy supports wildcarding). Use the logic here instead: https://github.com/istio/istio/blob/6cae0b991ad73742518914d95cdf7ac6f1060b1a/pilot/pkg/model/validation.go#L45-L46 

And here are some of the test cases:
https://github.com/istio/istio/blob/6cae0b991ad73742518914d95cdf7ac6f1060b1a/pilot/pkg/model/validation_test.go#L314-L323

Reply:
jfyi ... I've not written the tests for wildcard and CIDR.  I plan to do this once we have v2 on its feet.